sdk/cosmosdb/cosmos/src/client/ChangeFeed/ChangeFeedForEpkRange.ts

```diff
@@ -0,0 +1,383 @@
+  private continuationToken?: CompositeContinuationToken;
+  private queue: FeedRangeQueue<ChangeFeedRange>;
+  private startTime: string;
+  /**
+   * @internal
+   */
+  constructor(
+    private clientContext: ClientContext,
+    private partitionKeyRangeCache: PartitionKeyRangeCache,
+    private resourceId: string,
+    private resourceLink: string,
+    private url: string,
+    private rId: string,
+    private changeFeedOptions: InternalChangeFeedIteratorOptions,
+    private diagnosticContext: CosmosDiagnosticContext
+  ) {
+    super();
+
+    this.queue = new FeedRangeQueue<ChangeFeedRange>();
+    this.continuationToken = changeFeedOptions.continuationToken
+      ? JSON.parse(changeFeedOptions.continuationToken)
+      : undefined;
+    this.startTime = changeFeedOptions.startTime
+      ? changeFeedOptions.startTime.toUTCString()
+      : undefined;
+  }
+
+  private continuationTokenRidMatchContainerRid(): boolean {
+    if (this.continuationToken.rid !== this.rId) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Fetch all physical partitions for a given container.
+   */
+  public async fetchAllFeedRanges(): Promise<void> {
+    try {
+      const pkRanges = (
+        await this.partitionKeyRangeCache.onCollectionRoutingMap(this.url, this.diagnosticContext)
+      ).getOrderedParitionKeyRanges();
+      for (const pkRange of pkRanges) {
+        const feedRange: ChangeFeedRange = new ChangeFeedRange(
+          pkRange.minInclusive,
+          pkRange.maxExclusive,
+          ""
+        );
+        this.queue.enqueue(feedRange);
+      }
+    } catch (err) {
+      throw new Error(err.message);
+    }
+  }
+
+  /**
+   * Fetch all physical partitions overlapping the given Partition Key Range.
+   */
+  public async fetchOverLappingFeedRanges(pkRange: PartitionKeyRange | IEpkRange): Promise<void> {
+    const queryRange = new QueryRange(pkRange.minInclusive, pkRange.maxExclusive, true, false);
+
+    try {
+      const overLappingRanges = await this.partitionKeyRangeCache.getOverlappingRanges(
+        this.url,
+        queryRange,
+        this.diagnosticContext
+      );
+      for (const overLappingRange of overLappingRanges) {
+        const [epkMinHeader, epkMaxHeader] = await checkEpkHeaders(pkRange, overLappingRange);
+        const feedRange: ChangeFeedRange = new ChangeFeedRange(
+          overLappingRange.minInclusive,
+          overLappingRange.maxExclusive,
+          "",
+          epkMinHeader,
+          epkMaxHeader
+        );
+        this.queue.enqueue(feedRange);
+      }
+    } catch (err) {
+      throw new ErrorResponse(err.message);
+    }
+  }
+  /**
+   * Fill the queue with feed ranges from continuation token
+   */
+  public async fetchContinuationTokenFeedRanges(continuationToken: string): Promise<boolean> {
+    const contToken: CompositeContinuationToken = JSON.parse(continuationToken);
+    this.continuationToken = JSON.parse(continuationToken);
```

Comment:
Isn't this already happening on the constructor?

Reply:
Fixed this.